crates/graphql/src/schema/query_root.rs

```diff
@@ -441,6 +444,125 @@ impl QueryRoot {
+                        .map(ToString::to_string)
+                        .unwrap_or_default(),
+                    image: values
+                        .get("image")
+                        .and_then(Value::as_str)
+                        .map(ToString::to_string)
+                        .unwrap_or_default(),
+                    description: values
+                        .get("description")
+                        .and_then(Value::as_str)
+                        .map(ToString::to_string)
+                        .unwrap_or_default(),
+                    category: values
+                        .get("category")
+                        .and_then(Value::as_str)
+                        .map(ToString::to_string)
+                        .unwrap_or_default(),
+                    attributes: values
+                        .get("attributes")
+                        .and_then(Value::as_array)
+                        .map(|v| {
+                            v.iter()
+                                .map(|a| NftAttribute {
+                                    metadata_address: values
+                                        .get("id")
+                                        .map(ToString::to_string)
+                                        .unwrap_or_default(),
+                                    value: a
+                                        .get("value")
+                                        .map(ToString::to_string)
+                                        .unwrap_or_default(),
+                                    trait_type: a
+                                        .get("trait_type")
+                                        .map(ToString::to_string)
+                                        .unwrap_or_default(),
+                                })
+                                .collect::<Vec<NftAttribute>>()
+                        })
+                        .unwrap_or_default(),
+                }
+            })
+            .collect::<Vec<MetadataJson>>())
+    }
+
+    #[graphql(description = "returns profiles matching the search term")]
+    async fn profiles(
+        &self,
+        context: &AppContext,
+        #[graphql(description = "Search term")] term: String,
+        #[graphql(description = "Query limit")] limit: i32,
+        #[graphql(description = "Query offset")] offset: i32,
+    ) -> FieldResult<Vec<Wallet>> {
+        let search = &context.shared.search;
+
+        let result = search
+            .index("name_service")
+            .search()
+            .with_query(&term)
+            .with_offset(offset.try_into()?)
+            .with_limit(limit.try_into()?)
+            .execute::<Value>()
+            .await
+            .context("failed to load search result")?
+            .hits;
+
+        debug!("{:?}", result);
+
+        Ok(result
+            .into_iter()
+            .map(|r| {
+                let value = r.result;
+                Wallet {
```

Comment:
This should also just deserialize the value.

Reply:
yes but we would face the same problem of deserialization like we had with http metadata json indexing.Also, meilisearch query fails even if one of the item can not be deserialized